yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/scout_abilities.dm

```diff
@@ -0,0 +1,284 @@
+//////////////////////////////////////////////////////////////////////////
+//----------------------Scout light eater ability-----------------------//
+//////////////////////////////////////////////////////////////////////////
+/datum/action/cooldown/spell/toggle/light_eater
+	name = "Light Eater"
+	desc = "Twists an active arm into a blade of all-consuming shadow."
+	panel = "Darkspawn"
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
+	button_icon_state = "light_eater"
+	check_flags = AB_CHECK_HANDS_BLOCKED | AB_CHECK_CONSCIOUS | AB_CHECK_LYING
+	spell_requirements = SPELL_REQUIRES_DARKSPAWN | SPELL_REQUIRES_HUMAN
+
+/datum/action/cooldown/spell/toggle/light_eater/process()
+	active = owner.is_holding_item_of_type(/obj/item/light_eater)
+	. = ..()
+
+/datum/action/cooldown/spell/toggle/light_eater/can_cast_spell(feedback)
+	if(!owner.get_empty_held_indexes() && !active)
+		if(feedback)
+			to_chat(owner, span_warning("You need an empty hand for this!"))
+		return FALSE
+	. = ..()
+
+/datum/action/cooldown/spell/toggle/light_eater/Enable()
+	to_chat(owner, span_velvet("Akna"))
+	owner.visible_message(span_warning("[owner]'s arm contorts into a blade!"), span_velvet("You transform your arm into a blade."))
+	playsound(owner, 'yogstation/sound/magic/pass_create.ogg', 50, 1)
+	var/obj/item/light_eater/T = new(owner)
+	owner.put_in_hands(T)
+
+/datum/action/cooldown/spell/toggle/light_eater/Disable()
+	to_chat(owner, span_velvet("Haoo"))
+	owner.visible_message(span_warning("[owner]'s blade transforms back!"), span_velvet("You dispel the blade."))
+	playsound(owner, 'yogstation/sound/magic/pass_dispel.ogg', 50, 1)
+	for(var/obj/item/light_eater/T in owner)
+		qdel(T)
```

Comment:
The spell should hold reference to the light eater it makes so it wont just delete any others they mightve had on them